src/pykeen/nn/modules.py

```diff
@@ -1704,6 +1712,244 @@ def _prepare_state_for_functional(self) -> MutableMapping[str, Any]:  # noqa: D1
+
+        # compute width plus 1
+        widths_p1 = widths + 1
+
+        # compute box midpoints
+        # TODO: we already had this before, as `base`
+        centres = 0.5 * (box_lows + box_highs)
+
+        return torch.where(
+            # inside box?
+            torch.logical_and(points >= box_lows, points <= box_highs),
+            # yes: |p - c| / (w + 1)
+            torch.abs(points - centres) / widths_p1,
+            # no: (w + 1) * |p - c| - 0.5 * w * (w - 1/(w + 1))
+            widths_p1 * torch.abs(points - centres) - (0.5 * widths) * (widths_p1 - 1 / widths_p1),
+        )
+
+    @staticmethod
+    def boxe_kg_arity_position_score(
+        entity_pos: torch.FloatTensor,
+        other_entity_bump: torch.FloatTensor,
+        relation_box: Tuple[torch.FloatTensor, torch.FloatTensor],
+        tanh_map: bool,
+        p: int,
+        power_norm: bool,
+    ) -> torch.FloatTensor:
+        r"""Perform the BoxE computation at a single arity position.
+
+        .. note::
+            this computation is parallelizable across all positions
+
+        .. note ::
+            `entity_pos`, `other_entity_bump`, `relation_box_low` and `relation_box_high` have to be in broadcastable
+            shape.
+
+        :param entity_pos: shape: (*s_p, d)
+            This is the base entity position of the entity appearing in the target position. For example,
+            for a fact $r(h, t)$ and the head arity position, `entity_pos` is the base position of $h$.
+        :param other_entity_bump: shape: (*s_b, d)
+            This is the bump of the entity at the other position in the fact. For example, given a
+            fact $r(h, t)$ and the head arity position, `other_entity_bump` is the bump of $t$.
+        :param relation_box: shape: (*s_r, d)
+            The lower/upper corner of the relation box at the target arity position.
+        :param tanh_map:
+            whether to apply the tanh map regularizer
+        :param p:
+            The norm order to apply across dimensions to compute overall position score.
+        :param power_norm:
+            whether to use the powered norm instead
+
+        :return: shape: s
+            Arity-position score for the entity relative to the target relation box. Larger is better. the shape is the
+            broadcasted shape from position, bump and box, where the last dimension has been removed.
+        """
+        # Step 1: Apply the other entity bump
+        bumped_representation = entity_pos + other_entity_bump
+
+        relation_box_low, relation_box_high = relation_box
+
+        # Step 2: Apply tanh if tanh_map is set to True.
+        if tanh_map:
+            relation_box_low = torch.tanh(relation_box_low)
+            relation_box_high = torch.tanh(relation_box_high)
+            bumped_representation = torch.tanh(bumped_representation)
+
+        # Compute the distance function output element-wise
+        element_wise_distance = BoxEInteraction.point_to_box_distance(
```

Comment:
better to call this a classmethod so you can more idiomatically refer to other static methods in this class with `cls`